### ISSUE TITLE
The slice filter requires a mapping to operate on, got a <type 'list'>.

### ISSUE THREAD

**Issue by dobrozhanskyi** [ansible/ansible#37890]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

filter slice

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/invisible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### CONFIGURATION

```
ANSIBLE_PIPELINING(/etc/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=60s
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 20
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

MacOS 10.13.3

##### SUMMARY

fatal: [test]: FAILED! => {"changed": false, "msg": "AnsibleFilterError: The slice filter requires a mapping to operate on, got a <type 'list'>."}

##### STEPS TO REPRODUCE

In template use: 
{%- for column in items|slice(3) %}

```yaml

```

##### ACTUAL RESULTS

```
fatal: [test]: FAILED! => {"changed": false, "msg": "AnsibleFilterError: The slice filter requires a mapping to operate on, got a <type 'list'>."}
```

**Comment by dobrozhanskyi** [comment]
https://github.com/ansible/ansible/pull/35922/files

**Comment by alikins** [comment]
@dobrozhanskyi So this is a difference between the previous ansible 'slice' (removed in 90e267417b0a446fa825ab63ed4809a5e9a1b010) and the builtin jinja 'slice'?

**Comment by alikins** [comment]
pinging @bcoca re https://github.com/ansible/ansible/commit/90e267417b0a446fa825ab63ed4809a5e9a1b010

**Comment by bcoca** [comment]
the ansible 'slice' was never pushed, it only existed in /devel for a short time, until i realized I had missed the jinja2 already existed.

**Comment by alikins** [comment]
@dobrozhanskyi was this working in versions prior to 2.5.0? If so, what versions?

**Comment by alikins** [comment]
is the docs blurb at http://docs.ansible.com/ansible/2.5/user_guide/playbooks_filters.html# correct?

It sounds correct, even it if is actually referring to a jinja builtin feature. The 'new in version 2.5' is a little confusing in that context.

> The slice filter can be used to extract the values of specific keys from a hash:
> 
> {{ {'x': 1, 'y': 2, 'z': 3 } | slice(['x', 'z']) }}
>

**Comment by sivel** [comment]
> is the docs blurb at http://docs.ansible.com/ansible/2.5/user_guide/playbooks_filters.html# correct?

No. Since that filter never actually landed in 2.5, the docs should not be there either.

The slice filter in jinja2 is rather different http://jinja.pocoo.org/docs/dev/templates/#slice

**Comment by dobrozhanskyi** [comment]
It's working in ansible < 2.5. Example 2.4.3.
It's used in this role https://github.com/wtcross/ansible-sudoers/blob/44b813fc0a7817f3e4228374ccbbb22f2d4b3ee7/templates/sudoers_nospec.j2#L6 for example.

**Comment by sivel** [comment]
@bcoca it looks like the removal from stable-2.5 never happened.  It should not have been included in 2.5 but was: https://github.com/ansible/ansible/blob/stable-2.5/lib/ansible/plugins/filter/core.py#L491

2.5 must have been branched before you removed it.

**Comment by CafeLungo** [comment]
I am running into this now (with the wtcross.sudoers role). This needs to be reverted if it wasn't intended to be included.

**Comment by mathias-muench** [comment]
Ansible 2.5.1 still the same problem.  We are patching 'core.py' on all systems manually now.

**Comment by bcoca** [comment]
fixed via #37944, will be in 2.5.2
